pages/home/script.js

```diff
@@ -16,26 +17,155 @@ import geocode from '../../assets/helpers/geocode.js';
 import loadIcons from '../../assets/helpers/load-icons.js';
 //Models
 import Map from '../../assets/models/Map.js';
+import HazardReport from '../../assets/models/HazardReport.js';
+import getHazardDetail from '../../assets/helpers/get-hazard-detail.js';
+import safeNumber from '../../assets/helpers/safe-number.js';
+
+// URL params
+const url = new URL(window.location.href);
+const idReport = url.searchParams.get('id');
+const openDetail = url.searchParams.get('open') === 'true' && !!idReport;
+const focusMarker = url.searchParams.get('focus') === 'true' && !!idReport;
+const zoom = url.searchParams.get('zoom') ?? 4;
+const latitude = safeNumber(url.searchParams.get('lat'), null);
+const longitude = safeNumber(url.searchParams.get('lng'), null);
+
 //Variable Declaration
 let geoMap;
-let position = Map.DEFAULT_LOCATION;
-let positionObj = {};
-let locationDetails = '';
+let position =
+  latitude && longitude
+    ? { lat: latitude, lng: longitude }
+    : Map.DEFAULT_LOCATION;
+
+// let positionObj = {};
+// let locationDetails = '';
 let hazardCardParams = {};
 let searchSuggestions = [];
 let reports = [];
-let mapOptions = {
-  zoomControl: false,
-  doubleClickZoom: false,
-  CURRENT_ZOOM: 4,
-};
 let categoryFilters = [];
 let flyToTrigger = true;
 const FLY_TO_ZOOM = 12;
-const ANIMATION_DURATION = 4;
+// const ANIMATION_DURATION = 4;
 const alert = new AlertPopup();
 
-const categories = await apiRequest(`hazard-category`, { method: 'GET' });
+let mapOptions = {
+  zoomControl: false,
+  doubleClickZoom: false,
+  CURRENT_ZOOM: zoom,
+};
+
+let hazardDetail = new HazardReport();
+
+const searchBarParams = {
+  categories: [],
+};
+
+/**
+ * Page Init
+ */
+
+window.onload = async function () {
```

Comment:
Is `window.onload` needed here? Home page has no HTML content that has to be loaded before JS code execution

Reply:
Ya, I'm fine with it.

Reply:
I think "window.onload" is really useful here because it allows us to wait for the entire DOM to be fully ready, including HTML, JS, images, etc, not just the HTML

Reply:
I think "window.onload" is really useful here because it allows us to wait for the entire DOM to be fully ready, including HTML, JS, images, etc, not just the HTML